org.testeditor.web.dropwizard.xtext/src/main/java/org/testeditor/web/xtext/index/persistence/GradleProjectIndexUpdater.xtend

```diff
@@ -9,32 +9,49 @@ import java.nio.file.Paths
 import java.nio.file.StandardOpenOption
 import java.util.concurrent.TimeUnit
 import javax.inject.Inject
-import org.testeditor.web.xtext.index.CustomStandaloneBuilder
 import javax.inject.Singleton
+import org.eclipse.xtext.build.BuildRequest
+import org.eclipse.xtext.build.IncrementalBuilder
+import org.eclipse.xtext.resource.XtextResourceSet
+import org.eclipse.xtext.resource.impl.ResourceDescriptionsData
 import org.slf4j.LoggerFactory
+import org.testeditor.web.dropwizard.xtext.validation.ValidationMarkerUpdater
+
+import static extension org.eclipse.emf.common.util.URI.createFileURI
+import static extension org.testeditor.web.xtext.index.buildutils.XtextBuilderUtils.*
 
 @Singleton
 class GradleProjectIndexUpdater extends IndexUpdater {
-	
+
 	static val GRADLE_PROCESS_TIMEOUT_MINUTES = 10
 
 	static val logger = LoggerFactory.getLogger(GradleProjectIndexUpdater)
 
-	@Inject CustomStandaloneBuilder builder	
+	@Inject IncrementalBuilder builder
+	@Inject ValidationMarkerUpdater validationMarkerUpdater
 
 	override def void initIndex(File projectRoot) {
-		builder.languages = languageAccessors
-		if (new File(projectRoot.absolutePath, 'build.gradle').exists) {
+		val basePath = projectRoot.absolutePath
+
+		if (new File(basePath, 'build.gradle').exists) {
 			runGradleAssemble(projectRoot)
 			prepareGradleTask(projectRoot)
 			val jars = collectClasspathJarsViaGradle(projectRoot)
-			builder => [
-				baseDir = projectRoot.absolutePath
-				sourceDirs = #[baseDir + "/src/main/java", baseDir + '/src/test/java']
-				classPathEntries = jars + #[baseDir + '/build/classes/java/main']
+			val initRequest = new BuildRequest => [
+				baseDir = basePath.createFileURI
+				resourceSet = index.resourceSet as XtextResourceSet
```

Comment:
the index should probably provide a getter that returns this resourceSet as XtextResourceSet (will reduce the places where casts are used)